### PR TITLE
[12.0][MIG] account_asset_management

### PR DESCRIPTION
**Pull request by ghost**
TODO:
- Migration script for account.asset parent_path (replacing parent_left and parent_right)
- models/account_fiscal_year.py instead of models/date_range.py (I don't understand the purpose of this logic)

### CODE REVIEW

**Comment by ghost** [comment]
I have now signed both Individual CLA for @norlinhenrik and Entity CLA for @apps2grow.
I don't know why no github login is found for Stefan Rijnhard. He is not in my team...

**Comment by pedrobaeza** [comment]
@apps2grow don't worry about that one, as it's due to old emails being imported on commit history. We are tuning the CLA bot for that. See https://github.com/OCA/oca-github-bot/issues/28 for more details.

**Comment by ghost** [comment]
I would like to make a migration script for this module. I am learning by reading other migration scripts. But how to RUN the migration script?

**Comment by ghost** [comment]
Could this module merged? I want to translate it in german fully.

**Comment by tbaden** [comment]
@apps2grow 
to run a migration script you need a migrations directory, with a subdir with the new version_number of your module (e.g. 12.0.1.0.1). In there you can create your pre-migrate.py or post-migrate.py file. It'll run on module update, don't forget to set your version number correctly in the manifest.py

**Comment by sbidoul** [comment]
@apps2grow thanks for this work.

Regarding the fiscal year, IMO the dependency on OCA `account_fiscal_year` can be removed, and the search on the fiscal year `date.range` can be replaced by a search on the standard Odoo model `account.fiscal.year`.

About `parent_path` is a migration script necessary? @pedrobaeza can you confirm?

**Comment by sbidoul** [comment]
Or better, use `compute_fiscalyear_dates(self, current_date)` on res.company to find the fiscal year end date.

**Review by WhiteTiger08 — APPROVED** [review]

**Comment by randall-vx** [comment]
Hi @pedrobaeza ! What do you think about @sbidoul comments ? Regards !

**Comment by WhiteTiger08** [comment]
Any progress  here? may I help with something?

**Comment by pedrobaeza** [comment]
@apps2grow please include #828

**Comment by ghost** [comment]
How do I do it? I am new to this, but I would like to learn :)

**Comment by pedrobaeza** [comment]
You can use `git cherry-pick` with the hash of the commit in the PR. If several of them, just repeat the command with each.

**Comment by luc-demeyer** [comment]
I agree with @sbidoul to move from date_range to account.fiscal.year.
Concerning parent_id/child_ids : my intention is to remove this completely and move to tags for reporting purposes. I think we can already remove this now and work on the reporting module for Odoo 12.0 later on (e.g. OCA code sprint  during Odoo Experience).
https://github.com/OCA/account-financial-tools/pull/828 should indeed be included.

**Comment by WhiteTiger08** [comment]
What else are you missing here?

**Comment by Saran440** [comment]
Functional test
Case 1: Asset created directly -> salvage value is editable -> OK
Case 2: Asset created by supplier invoice -> salvage value is not editable -> Not OK

**Comment by kittiu** [comment]
I think many is expecting this important addons to be merged. Anything we can help?

**Comment by carlosdauden** [comment]
Wrong depreciation value when Computation Method is Linear up to Salvage Value and Prorata Temporis cheked.

http://3374499-756-f4d651.runbot3.odoo-community.org/web#id=2&action=254&model=account.asset&view_type=form&menu_id=89

Selección_233
Selección_234

Could be the problem here?:
https://github.com/OCA/account-financial-tools/blob/c507c7307639fed171196d5faac5d36af09c0bcc/account_asset_management/models/account_asset.py#L995-L1010

**Comment by luc-demeyer** [comment]
@Saran440 
can you open an issue for the problem with the not editable salvage value ?
This issue has no relation to the 12.0 port (and needs to be fixed also on the previous version of this module.

@carlosdauden 
same remark as for issue reported by @Saran440

**Comment by pedrobaeza** [comment]
@luc-demeyer but we can't merge this for being used in Spain instead of `account_asset` if it's not fixed...

**Comment by luc-demeyer** [comment]
@kittiu 
Question on : can we help ?
If you have time:
- removal of dependency on OCA account_fiscalyear
- removal of 'view' assets with parent_id (confusing, I already intended to do it during the 11.0 port but didn't find the time yet). This parent assets of type view are there only for reporting purposes. I propose to replace them by an optional field for m2m reporting tags. These tags should have a parent_id.
I think we should remove the confusing assets of type view before merging the 12.0 port. Implementing the tag based reports can be done in a later stage (my idea is to do this during the october OCA code sprint).

**Comment by luc-demeyer** [comment]
@pedrobaeza 
which one of the reported issues is critical for the spanish market and therefor blocking for a merge ?
In Belgium, I encountered the 'Linear up to Salvage Value' only once in the past 10 years. The issue reported by @Saran440 looks more important to me.

**Comment by pedrobaeza** [comment]
@luc-demeyer prorata temporis is always used in Spain, but now I'm seeing that the compute method is not the standard one, so if @carlosdauden confirms that the problem doesn't happen with other methods, it's OK for me.

**Comment by carlosdauden** [comment]
Ok.
I will open a new PR or Issue with that problem.

**Comment by luc-demeyer** [comment]
@carlosdauden 
if you open an issue then I'll fix it (and add a unit test) but since I am leaving on holiday tomorrow morning it may take a while.

**Comment by carlosdauden** [comment]
@luc-demeyer 
Issue opened.
This isn't too important, enjoy your holidays. :wink:

**Comment by sbidoul** [comment]
So is it correct the last two things to do here are
- [ ] remove the dependency on account_fiscal_year (should be trivial)
- [ ] cherry pick #838

**Comment by sbidoul** [comment]
@luc-demeyer IMO we should keep the view assets until the asset tags are implemented with a proper migration script from view assets to tags.

**Comment by ghost** [comment]
I did cherry pick #828 already, see in the end of the commit history.

File: account_asset_management/__manifest__.py
Comment (sbidoul) [inline account_asset_management/__manifest__.py]:
```suggestion
        'account',
```

**Comment by sbidoul** [comment]
@apps2grow #828 yes, I was mentioning #838.

**Comment by sbejaoui** [comment]
> last two things to do here are
> 
> * [ ]  remove the dependency on account_fiscal_year (should be trivial)
> * [ ]  cherry pick #838

Hi @apps2grow ,
I made a [PR](https://github.com/apps2grow/account-financial-tools/pull/1) to your source branch that resolve the two remaining tasks to finish this one.

Please consider review it and merge it if it looks good to you.

Best regards.

**Comment by ghost** [comment]
I have merged the PR from @sbejaoui

**Comment by luc-demeyer** [comment]
> @luc-demeyer IMO we should keep the view assets until the asset tags are implemented with a proper migration script from view assets to tags.

@sbidoul 
I agree with the principle but that is a bit more work and keeps the confusing User Experience caused by the view assets until we implement the reporting tags (I can take the lead on this during the october OCA code sprint). If we go that way, than we should also migrate xls reporting module (and hence a request if you could merge https://github.com/OCA/account-financial-tools/pull/701). I'll do the migration of the reporting module to 12.0 shortly after the merge (it should be primarily a matter of testing, hence not a lot of work).

**Comment by pedrobaeza** [comment]
I would like to have this merged ASAP, and of course the view assets are very confusing, but I can't push you to any path.

File: account_asset_management/models/__init__.py
Comment (luc-demeyer) [inline account_asset_management/models/__init__.py]: we should replace this import by "from . import account_fiscal_year", rename the date_range.py to account_fiscal_year.py and adjust the logic for table recompute when creating/changing fiscal year duration.

File: account_asset_management/models/account_asset.py
Comment (luc-demeyer) [inline account_asset_management/models/account_asset.py]: Fix https://github.com/OCA/account-financial-tools/commit/fc09774f7bf889b431e46f099d7d2eacdd6db498 to be cherrypicked.

**Comment by ghost** [comment]
I have updated the pull request now, with update in account_fiscal year.py and fix in account_asset.py.

**Comment by luc-demeyer** [comment]
@apps2grow 
Thanks a lot. I reviewed and tested the new code. Looks ok to me.
The only thing that needs to be done is to fix the few remaing flake8 issues and than we are ready for merge.

**Comment by ghost** [comment]
Done.

I need to migrate from https://github.com/odoo/odoo/tree/10.0/addons/account_asset to https://github.com/OCA/account-financial-tools/tree/10.0/account_asset_management in version 10.0 or 11.0. Does anyone have a migration script for this?

**Review by sbidoul — APPROVED** [review]

**Review by luc-demeyer — APPROVED** [review]

**Comment by ghost** [comment]
Great! How to replace account_asset with account_asset_management?

**Comment by luc-demeyer** [comment]
You'll have to develop a migration script.
You can have a look to https://github.com/OCA/account-financial-tools/tree/10.0/account_asset_management/migrations/8.0.3 and use this as a starting point, but it will be a bit of work to get a seemless migration from the odoo account_asset module to this one. I am also not sure if we want to have such a script as part of this module or rather have a separate module that does this.

**Comment by pedrobaeza** [comment]
It must inside this module, @luc-demeyer. It doesn't affect for new installations and people migrating from already installed `account_asset_management`. It should be detected in the migration script itself (something like `if not openupgrade.table_exists(...): return`

**Comment by luc-demeyer** [comment]
I assume we should use post_init_hook for this.

**Comment by pedrobaeza** [comment]
No, we should use regular migration mechanism + a module merge in OpenUpgrade, which I have just made there: https://github.com/OCA/OpenUpgrade/blob/4f1b58559ef590cca00a28adc5dc23aada0bd7be/odoo/addons/openupgrade_records/lib/apriori.py#L44

**Comment by sbidoul** [comment]
Shouldn't we provide a path for people who want to transition without a using openupgrade?

**Comment by pedrobaeza** [comment]
@sbidoul the path is to run manually the scripts in their DB. What other path can be?

**Comment by sbidoul** [comment]
post_init_hook as suggested by Luc? Could be dangerous though, I'm not sure.

**Comment by pedrobaeza** [comment]
That's not feasible, as you come from `account_asset` module installed, so you mustn't install also `account_asset_management` and then uninstall the other: you will lose data and mix references. That's why the path is to move all references from `account_asset` to `account_asset_management` and then move corresponding data between tables. If you are not using OpenUpgrade, then you must do it manually, using `openupgradelib` methods for updating that name... in a shell session for example.

**Comment by nikul-serpentcs** [comment]
IMP https://github.com/OCA/account-financial-tools/pull/852

**Pull request closed without merging**
